gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProc.java

```diff
@@ -35,38 +36,39 @@
 
 /**
  * Responsible for performing the actual work for a given {@link DagTask} by first initializing its state, performing
- * actions based on the type of {@link DagTask} and finally submitting an event to the executor.
+ * actions based on the type of {@link DagTask}. Submitting events in time is important (PR#3641), hence initialize and
```

Comment:
unless there's javadoc syntax I'm not familiar with, I'd suggest an actual link here - `https://github.com/apache/gobblin/pull/3896`